[AzSdkCli] Can no longer build due to NPM restrictions

```
C:\repos\azure-sdk-tools\tools\azsdk-cli>dotnet restore
Restore complete (1.2s)

Build succeeded in 1.3s

C:\repos\azure-sdk-tools\tools\azsdk-cli>dotnet build
Restore complete (0.8s)
  Azure.Sdk.Tools.Cli.Analyzer succeeded (0.2s) → C:\repos\azure-sdk-tools\artifacts\bin\Azure.Sdk.Tools.Cli.Analyzer\Debug\netstandard2.0\Azure.Sdk.Tools.Cli.Analyzer.dll
  Azure.Sdk.Tools.CodeownersUtils succeeded (0.3s) → C:\repos\azure-sdk-tools\artifacts\bin\Azure.Sdk.Tools.CodeownersUtils\Debug\net8.0\Azure.Sdk.Tools.CodeownersUtils.dll
  Azure.Sdk.Tools.Cli failed with 1 error(s) (0.5s)
    C:\.tools\.nuget\packages\github.copilot.sdk\1.0.8\build\GitHub.Copilot.SDK.targets(108,5): error MSB3923: Failed to download file "https://registry.npmjs.org/@github/copilot-win32-x64/-/copilot-win32-x64-1.0.73.tgz".  The SSL connection could not be established, see inner exception. ---> Authentication failed because the remote party sent a TLS alert: 'HandshakeFailure'. ---> The message received was unexpected or badly formatted.

Build failed with 1 error(s) in 1.9s
```

At the same time, I get this popup

<img width="369" height="206" alt="Image" src="https://github.com/user-attachments/assets/61a3254f-314e-463c-b0ee-a9950600fd37" />


While these are .NET dependencies, it seems the Github Copilot .NET SDK depends on or wraps the NPM package in some way.

I believe this is due to the recent 1ES change:


> Following our earlier rollout ensuring client devices have their NPM’s global defaults point to a CFS-protected internal feed, we’re rolling out new policy controls to block direct access to the NPM package registry.
> **TL;DR**
> Company-managed devices won’t be able to directly access the npmjs.org package registry using npm, pnpm, or yarn.
Your client tools should already be configured to pull from a CFS protected feed

--- a/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj
+++ b/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj
@@ -96,4 +96,21 @@
     <CopilotSkipCliDownload>true</CopilotSkipCliDownload>
   </PropertyGroup>
 
+  <!-- On managed devices where npmjs.org is blocked (e.g. 1ES policy), redirect
+       the Copilot CLI download to whatever registry npm is configured to use.
+       NPM_CONFIG_REGISTRY env var takes priority; if not set, we read from npm config. -->
+  <PropertyGroup Condition="'$(NPM_CONFIG_REGISTRY)' != ''">
+    <CopilotNpmRegistryUrl>$(NPM_CONFIG_REGISTRY)</CopilotNpmRegistryUrl>
+  </PropertyGroup>
+
+  <Target Name="_GetNpmRegistryUrl" BeforeTargets="_DownloadCopilotCli"
+          Condition="'$(CopilotSkipCliDownload)' != 'true' And '$(TF_BUILD)' != 'True'">
+    <Exec Command="npm config get registry" ConsoleToMSBuild="true" StandardOutputImportance="low" IgnoreExitCode="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="_NpmConfigRegistry" />
+    </Exec>
+    <PropertyGroup Condition="'$(_NpmConfigRegistry)' != '' And !$(_NpmConfigRegistry.Contains('registry.npmjs.org'))">
+      <CopilotNpmRegistryUrl>$(_NpmConfigRegistry.Trim())</CopilotNpmRegistryUrl>
+    </PropertyGroup>
+  </Target>
+
 </Project>

--- a/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj
+++ b/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj
@@ -104,11 +104,12 @@
   </PropertyGroup>
 
   <Target Name="_GetNpmRegistryUrl" BeforeTargets="_DownloadCopilotCli"
-          Condition="'$(CopilotSkipCliDownload)' != 'true' And '$(TF_BUILD)' != 'True'">
-    <Exec Command="npm config get registry" ConsoleToMSBuild="true" StandardOutputImportance="low" IgnoreExitCode="true">
+          Condition="'$(CopilotNpmRegistryUrl)' == '' And '$(CopilotSkipCliDownload)' != 'true' And '$(TF_BUILD)' != 'True'">
+    <Exec Command="npm config get registry" ConsoleToMSBuild="true" StandardOutputImportance="low" IgnoreExitCode="true" EchoOff="true">
       <Output TaskParameter="ConsoleOutput" PropertyName="_NpmConfigRegistry" />
+      <Output TaskParameter="ExitCode" PropertyName="_NpmExitCode" />
     </Exec>
-    <PropertyGroup Condition="'$(_NpmConfigRegistry)' != '' And !$(_NpmConfigRegistry.Contains('registry.npmjs.org'))">
+    <PropertyGroup Condition="'$(_NpmExitCode)' == '0' And '$(_NpmConfigRegistry)' != '' And !$(_NpmConfigRegistry.Contains('registry.npmjs.org'))">
       <CopilotNpmRegistryUrl>$(_NpmConfigRegistry.Trim())</CopilotNpmRegistryUrl>
     </PropertyGroup>
   </Target>

